test/functorch/test_aotdispatch.py

```diff
@@ -1542,6 +1542,39 @@ def assert_compiler(gm: torch.fx.GraphModule, _):
         res = aot_mod(*inputs)
         res[0].sum().backward()
 
+    def test_aot_module_simplified_fake_tensor_gm_raises(self):
+        class MockModule(torch.nn.Module):
+            def __init__(self, y):
+                super().__init__()
+                self.linear = torch.nn.Linear(4, 4)
+                self.y = y
+
+            def forward(self, x):
+                z = self.linear(x)
+                z = z + self.y
+                z = z.relu()
+                return (z, )
+
+
+        real_x = torch.randn(4)
+        real_y = torch.randn(4)
+        fake_mode = torch._subclasses.fake_tensor.FakeTensorMode()
+        fake_y = fake_mode.from_tensor(real_y)
+
+        tracer = torch.fx.Tracer()
+        tracer.record_stack_traces = True
+        graph = tracer.trace(MockModule(fake_y))
+        mod_fake = torch.fx.GraphModule(tracer.root, graph)
```

Comment:
I don't understand the test here. What exactly is erroneous about this code?

Some things I might have expected to see here:

1. Fakeifying a module and then passing it to `aot_module_simplified` (should be covered here)
2. A dynamo backend that returns a GraphModule with some fake buffers on it (doesn't look like it's covered here)
3. A bug in tracing that results in a fake tensor constant showing up in the traced joint graph (doesn't look like it's covered here). Probably easiest way to synthesize the bug is to pass a function that directly accesses a fake tensor from inside the tracing context.

Reply:
(3) is not covered in this PR - this is purely for the lowering component - we need a different PR for the joint graph because I think thats a little trickier. We could detect it on the bw_module, but I think it might be better to do it at a different level, where we raise up the buffers and params into inputs.

Reply:
Oh and 

>  What exactly is erroneous about this code?

self.y is fake here, which is illegal.

Reply:
> A dynamo backend that returns a GraphModule with some fake buffers on it (doesn't look like it's covered here)

Dynamo backends return functions tho, so while we could do this, the hook here would be clumsy.

Reply:
I don't think this level exists? If I add a constant into graphmodule from fx tracing, it always stays a constant, it never gets input-ified. I think we have to do some AOTAutograd specific test here; or maybe `make_fx`, but generic usage of `make_fx` COULD validly put fake tensors onto the returned GraphModule so you'd have to think about when exactly this invariant holds.

Reply:
Hmm, maybe I misunderstood you - are you suggesting we add this invariant to outputs? If so, I think this would be a new invariant. I think checking the inputs on the graph module covers the cases we ran into well enough, and I would rather keep this change conservative.

Reply:
> self.y is fake here, which is illegal.

Oh, I missed this. Would be a good idea to call it out in the test.

> Hmm, maybe I misunderstood you - are you suggesting we add this invariant to outputs? If so, I think this would be a new invariant. I think checking the inputs on the graph module covers the cases we ran into well enough, and I would rather keep this change conservative.

What I am saying is that when I get a GraphModule from make_fx, in the context of AOTAutograd this GraphModule should NEVER have fake tensors on it. This is NOT true for make_fx calls in general, but for Dynamo usage of AOTAutograd we do maintain this invariant.